pkg/client/oci/oci.go

```diff
@@ -0,0 +1,296 @@
+	// if err := r.Reload(); err != nil {
+	// 	return nil, errors.Trace(err)
+	// }
+	return r, nil
+}
+
+// List lists all chart names in a repo
+func (r *Repo) List() ([]string, error) {
+	return nil, errors.Errorf("list method is not supported yet")
+}
+
+// getTagManifest returns the manifests of a published tag
+func (r *Repo) getTagManifest(name, version string) (*ocispec.Manifest, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), getTimeout)
+	defer cancel()
+
+	u := *r.url
+	// Form API endpoint URL from repo url
+	u.Path = path.Join("v2", u.Path, name, "manifests", version)
+	req, err := http.NewRequestWithContext(ctx, "GET", u.String(), nil)
+	req.Header.Set("Accept", "application/vnd.oci.image.manifest.v1+json")
+
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if r.username != "" && r.password != "" {
+		req.SetBasicAuth(r.username, r.password)
+	}
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer resp.Body.Close()
+
+	status := resp.StatusCode
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Errorf("unexpected response — %d %q — from %s", status, http.StatusText(status), u.String())
+	}
+	if status != http.StatusOK {
+		return nil, errors.Errorf("unexpected response — %d %q, %s — from %s", status, http.StatusText(status), string(body), u.String())
+	}
+	tm := &ocispec.Manifest{}
+	if err := json.Unmarshal(body, tm); err != nil {
+		return nil, err
+	}
+	return tm, nil
+}
+
+// getChartDigest returns the digest of a published chart
+func (r *Repo) getChartDigest(name, version string) (string, error) {
+	tm, err := r.getTagManifest(name, version)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	for _, layer := range tm.Layers {
+		if layer.MediaType == "application/tar+gzip" {
+			return layer.Digest.String(), nil
+		}
+	}
+
+	return "", errors.NotFoundf("%s:%s digest", name, version)
+}
+
+// ListChartVersions lists all versions of a chart
+func (r *Repo) ListChartVersions(name string) ([]string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), getTimeout)
+	defer cancel()
+
+	u := *r.url
+	// Form API endpoint URL from repository URL
+	u.Path = path.Join("v2", u.Path, name, "tags", "list")
+
+	req, err := http.NewRequestWithContext(ctx, "GET", u.String(), nil)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if r.username != "" && r.password != "" {
+		req.SetBasicAuth(r.username, r.password)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer resp.Body.Close()
+
+	status := resp.StatusCode
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Errorf("unexpected response — %d %q — from %s", status, http.StatusText(status), u.String())
+	}
+
+	switch status {
+	case http.StatusNotFound:
+		// If status not found 404, it could mean the asset has no release yet
+		return []string{}, nil
```

Comment:
please add a TODO here to tackle it later: Use the `NotFound` error instead of just returning `nil` and handle that case in the caller, instead.